src/main/java/org/mozilla/zest/impl/HttpClientWrapper.java

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.zest.impl;
+
+import org.apache.commons.httpclient.*;
+import org.apache.commons.httpclient.auth.AuthScope;
+import org.apache.commons.httpclient.methods.*;
+import org.apache.commons.httpclient.params.HttpClientParams;
+import org.mozilla.zest.core.v1.ZestAuthentication;
+import org.mozilla.zest.core.v1.ZestHttpAuthentication;
+import org.mozilla.zest.core.v1.ZestRequest;
+import org.mozilla.zest.core.v1.ZestResponse;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.Locale;
+
+public class HttpClientWrapper implements ZestHttpClient{
```

Comment:
- Does it need to be public? Are there plans for other "external" classes to use/access it?
 - Could be called `CommonsHttpClient` instead, it's an implementation.

Reply:
#Does it need to be public? 
I think so, because if someone wants to change the behavior with `setParams()` or `setAuthentication()`  he must create an instance of this class and inject it in `ZestBasicRunner`

Reply:
They wouldn't need direct access to this class, they can use the existing methods (e.g. `setHttpClientParams`), maybe it would be easier to reuse this one if other changes are needed but the idea is to hide the use of Commons HttpClient to be able remove it sooner rather than later(?). One can always set a custom `ZestHttpClient` implementation if really needs to change how the messages are sent (or, the interface should be changed to expose that functionality if deemed general/useful for the majority?)

Reply:
> they can use the existing methods (e.g. setHttpClientParams, setHttpClient)
okay. But then we should not deprecate setHttpClientParams and setHttpClient until there is a replacement for Commons HttpClient, should't we?

Reply:
I think it's fine to deprecate those methods, it's now possible to set a custom `ZestHttpClient` (in case of ZAP it would just "redirect" to existing classes).